src/System.Linq.Queryable/src/System/Linq/EnumerableRewriter.cs

```diff
@@ -173,16 +168,25 @@ private Type GetEquivalentType(Type type)
                 }
                 if (equiv == null)
                 {
-                    var interfacesWithInfo = info.ImplementedInterfaces.Select(i => new { Type = i, Info = i.GetTypeInfo() }).ToArray();
+                    var interfacesWithInfo = info.ImplementedInterfaces.Select(IntrospectionExtensions.GetTypeInfo).ToArray();
                     var singleTypeGenInterfacesWithGetType = interfacesWithInfo
-                        .Where(i => i.Info.IsGenericType && i.Info.GenericTypeArguments.Length == 1)
-                        .Select(i => new { Type = i.Type, Info = i.Info, GenType = i.Info.GetGenericTypeDefinition() });
-                    Type typeArg = singleTypeGenInterfacesWithGetType.Where(i => i.GenType == typeof(IOrderedQueryable<>) || i.GenType == typeof(IOrderedEnumerable<>)).Select(i => i.Info.GenericTypeArguments[0]).Distinct().SingleOrDefault();
+                        .Where(i => i.IsGenericType && i.GenericTypeArguments.Length == 1)
+                        .Select(i => new {Info = i, GenType = i.GetGenericTypeDefinition()})
+                        .ToList();
```

Comment:
I see. In a common case (IEnumerable) it is used 2 times, so makes sense to snap it into a list. I missed that part.

---
In reply to: [89371299](https://github.com/dotnet/corefx/pull/13540#discussion_r89371299) [](ancestors = 89371299)